### DISCUSSION TITLE
[Chapter 08] `@GeneratedValue(strategy=GenerationType.AUTO)` 에 대한 의구점

### DISCUSSION THREAD

**Discussion by Irisation23** [cowaine/spring-boot-for-msa#93] [Chapter 08. Data Persistence] [answered]
`@GeneratedValue(strategy=GenerationType.AUTO)` 를 `@Id`의 전략으로 사용한다면 아래와 같은 설정이 Default가 된다.

```
Oracle을 사용한다면 시퀀스(GenerationType.SEQUENCE) 방식으로 매핑한다.
MySQL을 사용한다면 아이덴티티(GenerationType.IDENTITY) 방식으로 매핑한다.
```

### 왜 Oracle은 시퀀스를 기반으로 매핑되는 것이며, MySQL은 아이덴티티를 기반으로 매핑되는 것일까?

**Comment by ahngilwoong** [comment, marked as answer]
제 부서는 Oracle DB를 사용합니다.
Oracle DB는 MY SQL에 존재하는 AUTO_INCREMENT라는 기능이 없습니다.
그래서 오라클은 자동 키 생성 전략을 시퀀스라는 객체를 통하여 만들어 나가는 것으로 학습했습니다.
IDENTITY는 테이블에게 전략을 위임하는 건데
AUTO_INCREMENT가 테이블 단 동작이라고 합니다. 따라서 위의 두개가 나뉘는 것 같습니다.

**Comment by Irisation23** [comment]
### 질의 응답 이외의 CheckPoint

1. 답변자 안모씨는 현업 중 실제로 해당 문제 때문에 혼난적이 있다.
2. `@GeneratedValue(strategy=GenerationType.AUTO)` 를 이용하는 것이 DB 의존성을 낮출 수 있는 방법이라 생각함.
